src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/GetAzureContainerRegistryCredential.cs

```diff
@@ -16,43 +16,47 @@
 
 namespace Microsoft.Azure.Commands.ContainerRegistry
 {
-    [Cmdlet(VerbsCommon.Get, ContainerRegistryCredentialNoun,
-        DefaultParameterSetName = NameResourceGroupParameterSet),
-        OutputType(typeof(PSContainerRegistryCredential))]
+    [Cmdlet(VerbsCommon.Get, ContainerRegistryCredentialNoun)]
+    [OutputType(typeof(PSContainerRegistryCredential))]
     public class GetAzureContainerRegistryCredential : ContainerRegistryCmdletBase
     {
-        [Parameter(
-            Position = 0,
-            Mandatory = true,
-            ParameterSetName = NameResourceGroupParameterSet,
-            HelpMessage = "Resource Group Name.")]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
-        [Parameter(
-            Position = 1,
-            Mandatory = true,
-            ParameterSetName = NameResourceGroupParameterSet,
-            HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
         [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(
-            Mandatory = true,
-            ParameterSetName = RegistryObjectParameterSet,
-            ValueFromPipeline = true,
-            HelpMessage = "Container Registry Object.")]
-        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ParameterSetName = RegistryObjectParameterSet, HelpMessage = "Container Registry Object.")]
```

Comment:
As I mentioned that we can get the necessary info from ResourceId, why do we want to have two bindings?

Reply:
@Jinhuafei PowerShell first sees if it can bind the input object to a parameter _by value_, and if not, then it checks if it can bind the object to any of the parameters _by property name_. Even though the `PSContainerRegistry` object has a `ResourceId` property, the parameter set with the `InputObject` would have been the set used over the set with the `ResourceId`. The only difference in these two parameter sets is that the `Name` and `ResourceGroupName` are pulled from the properties of the object, where the `ResourceId` is parsed to get the `Name` and `ResourceGroupName`. This is a minor comment and I made it only for backwards compatibility, but after looking at it a little more, it doesn't seem to be necessary.

Reply:
@Jinhuafei `ValueFromPipeline` needs to be brought back